lib/apis/v3/felixconfig.go

```diff
@@ -87,6 +87,10 @@ type FelixConfigurationSpec struct {
 	// attempts to acquire the iptables lock if it is not available. Lower values make Felix more
 	// responsive when the lock is contended, but use more CPU. [Default: 50ms]
 	IptablesLockProbeInterval *metav1.Duration `json:"iptablesLockProbeInterval,omitempty" configv1timescale:"milliseconds" confignamev1:"IptablesLockProbeIntervalMillis"`
+	// IptablesFeatureDetectOverride is used to override the feature detection. Values are specified in
+	// a comma separated list. A value of "true" or "false" will force the feature, empty or omitted values
+	// are auto-detected. Features; SNATFullyRandom, MASQFullyRandom, RestoreSupportsLock
+	IptablesFeatureDetectOverride string `json:iptablesFeatureDetectOverride",omitempty"`
```

Comment:
Suggest making this just featureDetectOverride here too.

I think we need validation here to stop a user from setting a garbage value that would confuse felix.  Unfortunately, that means duplicating the function from felix.  Here's a sample commit that did some validation + tests: https://github.com/projectcalico/libcalico-go/commit/7f5b8c620b5caf2cf79e757d905e03e58db317c8#diff-5b3418236d048b3157da11cc90e65ad8